File: saxpy/openmp/nodataregion/saxpy.f90
program saxpy

  implicit none
  integer, parameter :: dp = selected_real_kind(15)
  integer, parameter :: ip = selected_int_kind(15)
  integer(ip) :: i,n
  real(dp),dimension(:),allocatable :: x, y
  real(dp) :: a,start_time, end_time

  n=500000000
  allocate(x(n),y(n))

  x = 1.0d0
  y = 2.0d0
  a = 2.0

  call cpu_time(start_time)
  do i = 1, n
     y(i) = y(i) + a * x(i)
  end do
  call cpu_time(end_time)
  deallocate(x,y)
  
  print '(a,f8.6)', 'SAXPY Time: ', end_time - start_time
  
end program saxpy
    

File: saxpy/openmp/solution/saxpy.c
#include <stdio.h>
#include <time.h>

int main() {
  long long int i, n=500000000;
  float a=2.0;
  float x[n];
  float y[n];
  clock_t start_time, end_time;

  for (i = 0; i < n; i++){
    x[i] = 1.0;
    y[i] = 2.0;
  }

  start_time = clock();

  for (i = 0; i < n; i++){
    y[i] = a*x[i] + y[i];
  }
  end_time = clock();
  printf ("SAXPY Time: %f\n", (double)(end_time - start_time)/CLOCKS_PER_SEC);
  
}
